### PR TITLE
Add an end-to-end test for the option to print compile jobs for module dependencies

### PR DESCRIPTION
**Pull request by artemcm**
As discovered by invoking the fast dependency scanner from within the driver.

### CODE REVIEW

File: Sources/SwiftDriver/Jobs/Planning.swift [hunk 1]
```diff
@@ -216,6 +210,18 @@ extension Driver {
     return jobs
   }
 
+  /// Prescan the source files to produce a module dependency graph and turn it into a set
+  /// of jobs required to build all dependencies.
+  public mutating func generatetExplicitModuleBuildJobs() throws -> [Job] {
```
Comment (DougGregor) [inline Sources/SwiftDriver/Jobs/Planning.swift:215]: Typo for "generate" in the function name.

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
The test in this PR invokes the Fast Dependency Scanner (with -frontend -scan-dependencies). I suspect the CI failure is caused by the fact that the CI machine's toolchain does not yet have a sufficiently recent version of the compiler and therefore does not have the Fast Dependency Scanner. :(

**Comment by DougGregor** [comment]
> The test in this PR invokes the Fast Dependency Scanner (with -frontend -scan-dependencies). I suspect the CI failure is caused by the fact that the CI machine's toolchain does not yet have a sufficiently recent version of the compiler and therefore does not have the Fast Dependency Scanner. :(

@shahmishal what is the process by which the toolchain gets updated for CI on this repo? We're doing some iteration with the main `swift` repo

**Comment by shahmishal** [comment]
It's using released compiler - Swift 5.2, I can update the job to use master nightly if that helps.

**Comment by artemcm** [comment]
That would help a lot for this and upcoming changes.

**Comment by DougGregor** [comment]
> It's using released compiler - Swift 5.2, I can update the job to use master nightly if that helps.

That would be wonderful, thank you!

**Comment by DougGregor** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Comment by DougGregor** [comment]
@swift-ci test

**Comment by artemcm** [comment]
As I kept iterating on this, this PR and the test it adds got subsumed into: #113

**Pull request closed without merging**